client/src/components/TodoList.vue

```diff
@@ -9,27 +9,53 @@
       </div>
     </div>
      <!-- // we are now passing the data to the todo component to render the todo list -->
-    <todo v-on:delete-todo="deleteTodo" v-on:toggle-todo="toggleTodo" v-for="todo in todos" v-bind:todo="todo" :key="todo.id">></todo>
+    <todo v-on:delete-todo="deleteTodo" v-on:toggle-todo="toggleTodo" v-on:edit-todo="editTodo" v-for="todo in todos" v-bind:todo="todo" :key="todo.id">></todo>
   </div>
 </template>
 
 <script type = "text/javascript" >
-
+import axios from 'axios';
 import Todo from './Todo';
 
+const url = 'http://localhost:9393';
+
 export default {
   props: ['todos'],
   components: {
     Todo,
   },
   methods: {
     deleteTodo(todo) {
-      const todoIndex = this.todos.indexOf(todo);
-      this.todos.splice(todoIndex, 1);
+      axios.delete(`${url}/todo/${todo.id}`)
+      .then((response) => {
+        if (response.data.success === 'ok') {
```

Comment:
This step is not necessary, if the response was a `2**` status code then the deletion was successful, you can change the server response to just `halt 204`

Reply:
is the unnecessary step the `if` response check?

Reply:
ok